docs/requirements.md

```diff
@@ -0,0 +1,23 @@
+# Requirements
+
+* The API must have zero dependencies and be header-only.
+
+Pros: allows you to drop it into a source tree (i.e. vendor it) without changing
+the build or adding a dependency.
+
+* The SDK should have minimal dependencies.
+
+* Exporters can bring in transport dependencies, e.g. Stackdriver exporter
+depends on gRPC.
+
+* Support both static linking and dynamic loading.
+
+Dynamic loading is required because some vendors want to drop in their own
+implementation of the SDK. It also allows low-dependency builds.
+
+This also requires a stable C++ ABI. There is quite a bit of interest in this.
+
+This means STL types can't be used in the interface.
```

Comment:
Or only STL types with well defined ABIs (e.g. std::array, std::pair)

Reply:
I'm a little bit worried because e.g. libc++ and absl can't even agree on string_view. :)

I would prefer nostd::span (i.e. array_view) over std::array.

As for pair IDK. Are you sure it won't cause a problem in future?